src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -0,0 +1,122 @@
+import jewellery.inventory.exception.product.ProductOwnerEqualsRecipientException;
+import jewellery.inventory.exception.product.ProductOwnerNotSeller;
+import jewellery.inventory.helper.SaleTestHelper;
+import jewellery.inventory.mapper.SaleMapper;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.Sale;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Resource;
+import jewellery.inventory.repository.*;
+import jewellery.inventory.service.SaleService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SaleServiceTest {
+  @InjectMocks private SaleService saleService;
+  @Mock private SaleRepository saleRepository;
+  @Mock private UserRepository userRepository;
+  @Mock private ProductRepository productRepository;
+  @Mock private SaleMapper saleMapper;
+  private User seller;
+  private User buyer;
+  private Product product;
+  private Sale sale;
+  private SaleRequestDto saleRequestDto;
+  private SaleRequestDto saleRequestDtoOwnerEqualsRecipient;
+  private SaleResponseDto saleResponseDto;
+  private ProductPriceDiscountRequestDto productPriceDiscountRequestDto;
+  private List<SaleResponseDto> saleResponseDtoList;
+  private List<Product> productsForSale;
+
+  @BeforeEach
+  void setUp() {
+    seller = createTestUserForSale();
+    buyer = createSecondTestUser();
+    product = getTestProduct(seller, new Resource());
+    productsForSale = SaleTestHelper.getProList(product);
+    sale = SaleTestHelper.createSaleWithTodayDate(seller, buyer, productsForSale);
+    saleResponseDto = SaleTestHelper.getSaleResponseDto(sale);
+    productPriceDiscountRequestDto =
+        SaleTestHelper.createProductPriceDiscountRequest(product.getId(), 1000, 10);
+    List<ProductPriceDiscountRequestDto> productPriceDiscountRequestDtoList = new ArrayList<>();
+    productPriceDiscountRequestDtoList.add(productPriceDiscountRequestDto);
+    saleRequestDto =
+        SaleTestHelper.createSaleRequest(
+            seller.getId(), buyer.getId(), productPriceDiscountRequestDtoList);
+    saleRequestDtoOwnerEqualsRecipient =
+            SaleTestHelper.createSaleRequest(
+                    seller.getId(), seller.getId(), productPriceDiscountRequestDtoList);
+    saleResponseDtoList = SaleTestHelper.getSaleResponseList(saleResponseDto);
+  }
+
+  @Test
+  void testGetAllSales() {
+
+    List<Sale> sales = Arrays.asList(sale, new Sale(), new Sale());
+
+    when(saleRepository.findAll()).thenReturn(sales);
+
+    List<SaleResponseDto> responses = saleService.getAllSales();
+
+    Assertions.assertEquals(sales.size(), responses.size());
+  }
+
+  @Test
+  void testCreateSaleProductWhenDateIsCorrect() {
```

Comment:
The test names should be following a specific pattern:
will<action>Successfully - happy path
will<action>Throws<Exception>When<condition> - unhappy path with reason.

I see in our code base that not all of them are following this pattern but we should start somewhere